### PR TITLE
Re-Image Functionality Now Includes 'Upgrade' Call

### PR DESCRIPTION
**Pull request by nharper285**
## Summary of the Pull Request

_What is this about?_
Currently, OneFuzz re-images VMSS nodes when jobs are complete and after 7 days. I've included a new call to begin_update_instances() which 'upgrades' nodes before they re-image. 

## PR Checklist
* [ ] Applies to work item: #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_
Changes to vmss.py. 

## Validation Steps Performed

_How does someone test & validate?_
I'm manually testing to see if a node is upgraded before it's reimaged. I've done this test by changing the timer from 7 days to 1 hour. 
* [ ] Re-Imaged VMSS node is also upgraded.

### CODE REVIEW

File: src/api-service/__app__/onefuzzlib/workers/scalesets.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
 from .shrink_queue import ShrinkQueue
 
 NODE_EXPIRATION_TIME: datetime.timedelta = datetime.timedelta(hours=1)
-NODE_REIMAGE_TIME: datetime.timedelta = datetime.timedelta(days=7)
+NODE_REIMAGE_TIME: datetime.timedelta = datetime.timedelta(hours=1)
```
Comment (mgreisen) [inline src/api-service/__app__/onefuzzlib/workers/scalesets.py:47]: Why this change?

Comment (nharper285) [reply src/api-service/__app__/onefuzzlib/workers/scalesets.py:47]: Temporary for testing!

File: src/api-service/__app__/onefuzzlib/azure/vmss.py [hunk 2]
```diff
@@ -180,6 +180,11 @@ def reimage_vmss_nodes(name: UUID, vm_ids: Set[UUID]) -> Optional[Error]:
             logging.info("unable to find vm_id for %s:%s", name, vm_id)
 
     if instance_ids:
+        compute_client.virtual_machine_scale_sets.begin_update_instances(
```
Comment (chkeita) [inline src/api-service/__app__/onefuzzlib/azure/vmss.py:183]: Does the `reimage` gets queued up if there is an `update` in progress? 
I am concerned that the second operation my fail unless we wait for the result of the first.

**Review by chkeita — APPROVED** [review]

File: src/api-service/__app__/onefuzzlib/azure/vmss.py (hunk 2 above)
Comment (mgreisen) [reply src/api-service/__app__/onefuzzlib/azure/vmss.py:183]: Adding a comment here for the record. Yes, according to the VMSS team.

Comment (ranweiler) [reply src/api-service/__app__/onefuzzlib/azure/vmss.py:183]: I had the same question! I suggest we add a comment in the code documenting this expectation (and why we have it).

Comment (ranweiler) [reply src/api-service/__app__/onefuzzlib/azure/vmss.py:183]: Some logging here might be helpful, too.

**Review by ranweiler — APPROVED** [review]

File: src/api-service/__app__/onefuzzlib/azure/vmss.py (hunk 2 above)
Comment (nharper285) [reply src/api-service/__app__/onefuzzlib/azure/vmss.py:183]: Cheick and I chatted about this yesterday. I'll make sure to follow-up on the comment anyways next time. 

I'll add a comment _and_ logging, because I agree it'll be helpful down the road.

Comment (nharper285) [reply src/api-service/__app__/onefuzzlib/azure/vmss.py:187]: @ranweiler - I added a comment and logging.

**Review by mgreisen — APPROVED** [review]
